addons/search-profile/userscript.js

```diff
@@ -7,19 +7,36 @@ export default async function ({ addon, console, msg }) {
     user = document.querySelector('[name="q"]').value.trim(),
     valid = /^[\w-]{3,20}$/g.test(user);
   tab.type = "button";
+  tab.classList.add("sa-search-profile-btn");
   tab.setAttribute("role", "tab");
   tab.setAttribute("aria-selected", false);
   tab.tabIndex = -1; // unselected tabs should only be focusable using arrow keys
   img.src = addon.self.dir + "/user.svg";
-  img.className = "tab-icon";
+  img.className = "tab-icon sa-search-profile-icon";
   span.innerText = msg("profile");
   addon.tab.displayNoneWhileDisabled(tab);
 
+  const setInvalidUsername = () => {
+    tab.disabled = true;
+    tab.title = msg("invalid-username", { username: user });
+  };
+
   // Check if a valid username is entered
   if (valid) {
     tab.addEventListener("click", () => {
       location = `/users/${user}/`;
     });
+    fetch(`https://api.scratch.mit.edu/users/${user}`)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.code == "NotFound") {
+          setInvalidUsername();
+        } else {
```

Comment:
There could be an error code other than `NotFound`.
```suggestion
        if (data.code == "NotFound") {
          setInvalidUsername();
        } else if (!data.code) {
```